packages/world/ts/recipes.ts

```diff
@@ -281,24 +286,1594 @@ export const recipes: Recipe[] = [
+    outputs: [["LightBlueCotton", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Cotton", 8],
+      ["LightGrayDye", 1],
+    ],
+    outputs: [["LightGrayCotton", 8]],
+  },
+
+  // Colored terracotta recipes
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["BrownDye", 1],
+    ],
+    outputs: [["BrownTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["OrangeDye", 1],
+    ],
+    outputs: [["OrangeTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["WhiteDye", 1],
+    ],
+    outputs: [["WhiteTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["YellowDye", 1],
+    ],
+    outputs: [["YellowTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["RedDye", 1],
+    ],
+    outputs: [["RedTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["CyanDye", 1],
+    ],
+    outputs: [["CyanTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["BlackDye", 1],
+    ],
+    outputs: [["BlackTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["PurpleDye", 1],
+    ],
+    outputs: [["PurpleTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["BlueDye", 1],
+    ],
+    outputs: [["BlueTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["MagentaDye", 1],
+    ],
+    outputs: [["MagentaTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["LightGrayDye", 1],
+    ],
+    outputs: [["LightGrayTerracotta", 8]],
+  },
+  {
+    station: "Workbench",
+    inputs: [
+      ["Terracotta", 8],
+      ["LightBlueDye", 1],
+    ],
+    outputs: [["LightBlueTerracotta", 8]],
+  },
+  // Functional objects recipes
+  {
+    station: "Workbench",
+    inputs: [["Glass", 3]],
+    outputs: [["GlassPane", 8]],
+  },
+  {
+    inputs: [["OakPlanks", 2]],
+    outputs: [["OakDoor", 1]],
+  },
+  {
+    inputs: [["BirchPlanks", 2]],
+    outputs: [["BirchDoor", 1]],
+  },
+  {
+    inputs: [["JunglePlanks", 2]],
+    outputs: [["JungleDoor", 1]],
+  },
+  {
+    inputs: [["SakuraPlanks", 2]],
+    outputs: [["SakuraDoor", 1]],
+  },
+  {
+    inputs: [["AcaciaPlanks", 2]],
+    outputs: [["AcaciaDoor", 1]],
+  },
+  {
+    inputs: [["SprucePlanks", 2]],
+    outputs: [["SpruceDoor", 1]],
+  },
+  {
+    inputs: [["DarkOakPlanks", 2]],
+    outputs: [["DarkOakDoor", 1]],
+  },
+  {
+    inputs: [["MangrovePlanks", 2]],
+    outputs: [["MangroveDoor", 1]],
+  },
+  {
+    inputs: [["IronBar", 2]],
+    outputs: [["IronDoor", 1]],
+  },
+  {
+    inputs: [["OakPlanks", 3]],
+    outputs: [["OakTrapdoor", 1]],
```

Comment:
Do we have support for trapdoors in the client @dhvanipa?

Reply:
no dont think theres trapdoors